src/components/Shared/Button/Button.js

```diff
@@ -1,17 +1,34 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import classnames from 'classnames';
+import FontAwesome from 'react-fontawesome';
 
 import './Button.css';
 
-export default function Button({ onClick, className, children, ...props }) {
+export default function Button({
+  onClick,
+  className,
+  children,
+  loading,
+  disabled,
+  ...props
+}) {
   return (
     <button
       className={classnames('Shared__button', className)}
       onClick={onClick}
+      disabled={!!(loading || disabled)}
       {...props}
     >
-      {children}
+      {loading && (
+        <FontAwesome
+          name="spinner"
+          size="lg"
+          className="fa-spin Shared__button-spinner"
+        />
+      )}
+      {loading && 'Loading'}
```

Comment:
Is there a reason to have this split up instead of having the spinner and 'Loading' string in in the same && check?

Reply:
Nope